src/swsssdk/port_util.py

```diff
@@ -57,15 +57,18 @@ def get_index_from_str(if_name):
         if match:
             return int(match.group(1)) + baseidx
 
-def get_interface_oid_map(db):
+def get_interface_oid_map(db, blocking=True):
     """
         Get the Interface names from Counters DB
     """
     db.connect('COUNTERS_DB')
-    if_name_map = db.get_all('COUNTERS_DB', 'COUNTERS_PORT_NAME_MAP', blocking=True)
-    if_lag_name_map = db.get_all('COUNTERS_DB', 'COUNTERS_LAG_NAME_MAP', blocking=True)
+    if_name_map = db.get_all('COUNTERS_DB', 'COUNTERS_PORT_NAME_MAP', blocking)
+    if_lag_name_map = db.get_all('COUNTERS_DB', 'COUNTERS_LAG_NAME_MAP', blocking)
     if_name_map.update(if_lag_name_map)
 
+    if if_name_map == {}:
```

Comment:
```suggestion
    if if_name_map:
```

Reply:
The code written in lines 69 and 70 is an enhancement. 
If we run on a system without ports (where if_name_map={}) there is no point to execute the rest of the function. 
In line 69 we check if the dictionary is empty and if so we return.
The proposed change is simply not doing what we want to do (it is not checking that the dictionary is empty).

Reply:
Sorry for misleading. My suggestion is actually mistakenly written. I updated.
I totally understand your point. My comment is suggestion to treat empty and None the same way, so it will tolerate the implementation detail of `db.get_all`

Reply:
Done.